pedsnetdcc/tests/prepdb_test.py

```diff
@@ -0,0 +1,239 @@
+import os
+import re
+import unittest
+
+import psycopg2
+
+from pedsnetdcc.utils import make_conn_str
+from pedsnetdcc.prepdb import (prepare_database, _make_database_name,
+                               _conn_str_with_database, _sites_and_dcc)
+
+# Huh, Python 3 is good for something; it has re.fullmatch
+if not hasattr(re, 'fullmatch'):
+    # http://stackoverflow.com/a/30212799/390663
+    def fullmatch(regex, string, flags=0):
+        """Emulate python-3.4 re.fullmatch()."""
+        return re.match("(?:" + regex + r")\Z", string, flags=flags)
+    re.fullmatch = fullmatch
+
+
+def introspect_schemas(conn_str):
+    """Return list of schemas in the database
+
+    A future version of this for more generic testing might produce
+    a nested data structure with tables under each schema ....
+
+    :param conn_str: libpq/psycopg2 connection string
+    :type: str
+    :return: list of schemas
+    :rtype: list(str)
+    """
+    schemas = []
+    exclude_schemas = (
+        'pg_catalog',
+        'pg_toast.*',
+        'pg_temp.*',
+        'information_schema'
+    )
+
+    with psycopg2.connect(conn_str) as conn:
+        with conn.cursor() as cursor:
+
+            cursor.execute(
+                'select schema_name from information_schema.schemata')
+            for row in cursor.fetchall():
+                schema = row[0]
+                should_exclude = False
+                for pat in exclude_schemas:
+                    if re.fullmatch(pat, schema):
+                        should_exclude = True
+                        break
+                if not should_exclude:
+                    schemas.append(row[0])
+    conn.close()
+
+    return schemas
+
+
+def exec_ddl(conn_str, sql, no_transaction=False):
+    with psycopg2.connect(conn_str) as conn:
+        if no_transaction:
+            conn.set_isolation_level(0)
+        with conn.cursor() as cursor:
+            cursor.execute(sql)
+    conn.close()
+
+
+class TestPrepareDatabase(unittest.TestCase):
+
+    roles = ('peds_staff', 'harvest_user', 'achilles_user', 'dqa_user',
+             'pcor_et_user', 'loading_user', 'dcc_owner', 'pcornet_sas')
+
+    def create_roles(self):
+        """Create roles so grants will not fail.
```

Comment:
Yes, I thought about the same thing and left it this way just because the dbhi/pedsnet scripts had already staked out that turf. But it would be best to gather everything under `pedsnetdcc`, in which case, either `prepdb` could subsume the role creation and handle errors when roles already exist, or it could invoke a role creation method in a possible `createcluster` module.